src/main/java/rx/internal/operators/OperatorStringSplit.java

```diff
@@ -0,0 +1,156 @@
+package rx.internal.operators;
+
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.regex.Pattern;
+
+import rx.Observable.Operator;
+import rx.Producer;
+import rx.Subscriber;
+import rx.functions.Action0;
+import rx.subscriptions.Subscriptions;
+
+public class OperatorStringSplit implements Operator<String, String> {
+    private final Pattern pattern;
+
+    public OperatorStringSplit(final Pattern pattern) {
+        this.pattern = pattern;
+    }
+
+    @Override
+    public Subscriber<? super String> call(Subscriber<? super String> child) {
+        StringSplitSubscriber cs = new StringSplitSubscriber(child, pattern);
+        child.add(cs);
+        StringSplitProducer cp = new StringSplitProducer(cs);
+        child.setProducer(cp);
+        return cs;
+    }
+
+    static final class StringSplitProducer implements Producer {
+        private StringSplitSubscriber cs;
+
+        public StringSplitProducer(StringSplitSubscriber cs) {
+            this.cs = cs;
+        }
+
+        @Override
+        public void request(long n) {
+            cs.requestFromChild(n);
+        }
+    }
+
+    static final class StringSplitSubscriber extends Subscriber<String> {
+        private final NotificationLite<String> nl = NotificationLite.instance();
+
+        private final Pattern pattern;
+        private final ConcurrentLinkedQueue<Object> queue;
+
+        private final AtomicLong requested = new AtomicLong();
+        private final Subscriber<? super String> child;
+
+        public StringSplitSubscriber(Subscriber<? super String> s, Pattern pattern) {
+            super(s);
+            this.pattern = pattern;
+            this.queue = new ConcurrentLinkedQueue<Object>();
+            this.child = s;
+
+            add(Subscriptions.create(new Action0() {
+                @Override
+                public void call() {
+                    queue.clear();
+                }
+            }));
+        }
+
+        private void requestFromChild(long n) {
+            if (n <= 0)
+                return;
+            // we track 'requested' so we know whether we should subscribe the next or not
+
+            long previous = BackpressureUtils.getAndAddRequest(requested, n);
+            if (previous == 0) {
+                request(2);
```

Comment:
Also copied from the original concat.
```
        public void onStart() {
            // no need for more than 1 at a time since we concat 1 at a time, so we'll request 2 to start ...
            // 1 to be subscribed to, 1 in the queue, then we'll keep requesting 1 at a time after that
            request(2);
        }
```